src/Formats/FormatFactory.cpp

```diff
@@ -207,6 +207,12 @@ FormatSettings getFormatSettings(const ContextPtr & context, const Settings & se
     format_settings.parquet.use_custom_encoder = settings[Setting::output_format_parquet_use_custom_encoder];
     format_settings.parquet.parallel_encoding = settings[Setting::output_format_parquet_parallel_encoding];
     format_settings.parquet.data_page_size = settings[Setting::output_format_parquet_data_page_size];
+    if (settings[Setting::output_format_parquet_batch_size] == 0)
+    {
+        throw DB::Exception(
+            ErrorCodes::BAD_ARGUMENTS,
+            "Invalid value for setting 'output_format_parquet_batch_size': must be greater than 0");
+    }
```

Comment:
Instead of explicit check here we can just use type `NonZeroUInt64` for this setting in `FormatFactorySettings.h`. The check will happend when we set the setting.